proto/structured.proto

```diff
@@ -104,3 +105,220 @@ message CreateTableResponse {
+  // the first request should set this field to non-nil with name and
+  // isolation level set as desired. The response will contain the
+  // fully-initialized transaction with txn ID, priority, initial
+  // timestamp, and maximum timestamp. The transaction returned in a
+  // response is feed to the next request.
+  optional Transaction txn = 9;
+  // ReadConsistency specifies the consistency for read
+  // operations. The default is CONSISTENT. This value is ignored for
+  // write operations.
+  optional ReadConsistencyType read_consistency = 10 [(gogoproto.nullable) = false];
+}
+
+// TableResponseHeader is returned with every structured API response.
+message TableResponseHeader {
+  // Error is non-nil if an error occurred.
+  optional Error error = 1;
+  // Timestamp specifies time at which read or write actually was
+  // performed. In the case of both reads and writes, if the timestamp
+  // supplied to the request was 0, the wall time of the node
+  // servicing the request will be set here. Additionally, in the case
+  // of writes, this value may be increased from the timestamp passed
+  // with the RequestHeader if the key being written was either read
+  // or written more recently.
+  //
+  // TODO(vivek): Is this only relevant in the non-transactional case?
+  optional Timestamp timestamp = 2 [(gogoproto.nullable) = false];
+  // Transaction is non-nil if the request specified a non-nil
+  // transaction. The transaction timestamp and/or priority may have
+  // been updated, depending on the outcome of the request.
+  optional Transaction txn = 3;
+}
+
+// A TableKey represents a key into a table index.
+message TableKey {
+  optional string table_name = 1;
+  // A column key component. Multiple ColumnKeys together form a key.
+  message ColumnKey {
+    optional string column_name = 1;
+    optional bytes key = 2; 
+  }
+  // The different key components from each column.
+  repeated ColumnKey columns = 2;
+}
+
+// A TableRow represents a set of column values within a table row. 
+message TableRow {
+  message Cell {
+    optional bytes column_name = 1;
+    optional Value value = 2;
+  }
+  repeated Cell cells = 1;
+}
+
+// A GetTableRequest is used to read a table row.
+message GetTableRequest {
+  optional TableRequestHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  optional TableKey key = 2 [(gogoproto.nullable) = false];
+}
+
+// A GetTableResponse returns a row.
+// If the key doesn't exist, it returns an empty row.
+message GetTableResponse {
+  optional TableResponseHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  optional TableRow row = 2 [(gogoproto.nullable) = false];
+}
+
+// A PutTableRequest inserts/updates a row in the table. To write
+// an empty value in a cell, a cell with a value are still specified, but both
+// Bytes and Integer in the Value are set to nil.
+message PutTableRequest {
```

Comment:
TableRow is the scope being operated on. I've added a comment about how a user uses Put to delete a specific cell. To delete an entire row users should use DeleteTableRowRequest